src/data/finset/basic.lean

```diff
@@ -2504,6 +2504,79 @@ cons_eq_insert _ _ h ▸ to_list_cons _
 
 end to_list
 
+/-!
+### disj_Union
+
+This section is about the bounded union of a disjoint indexed family `t : α → finset β` of finite
+sets over a finite set `s : finset α`. In most cases `finset.bUnion` should be preferred.
+-/
+section disj_Union
+
+variables {s s₁ s₂ : finset α} {t t₁ t₂ : α → finset β}
+
+/-- `disj_Union s f h` is the set such that `a ∈ disj_Union s f` iff `a ∈ f i` for some `i ∈ s`.
+It is the same as `s.bUnion f`, but it does not require decidable equality on the type. The
+hypothesis ensures that the sets are disjoint. -/
+def disj_Union (s : finset α) (t : α → finset β)
+  (hf : (s : set α).pairwise $ λ a b, ∀ x, x ∈ t a → x ∉ t b) : finset β :=
```

Comment:
Once #16434 has gone through, you could use the following, right?
```suggestion
  (hf : (s : set α).pairwise_disjoint t) : finset β :=
```
That sounds desirable, so maybe we should make them dependent.

Reply:
Yes, that was one of the motivations for #16434; I think it makes sense for #16434 to depend on this PR, since it will be a much slower PR.